compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2281,6 +2281,121 @@ impl<'tcx> Ty<'tcx> {
             _ => None,
         }
     }
+
+    pub fn is_c_void(self, tcx: TyCtxt<'_>) -> bool {
+        match self.kind() {
+            ty::Adt(adt_def, ..) => {
+                let def_id = adt_def.0.did;
+                let crate_name = tcx.crate_name(def_id.krate);
+                if adt_def.is_enum()
+                    && tcx.item_name(def_id) == sym::c_void
+                    && (crate_name == sym::core
+                        || crate_name == sym::std
+                        || crate_name == sym::libc)
```

Comment:
Starting from rustc 1.30, `libc::c_void` is merely a re-export of `core::ffi::c_void`. The libstd one is also a re-export.

Reply:
Done.